contracts/test/ChainLinkOracle.t.sol

```diff
@@ -1,42 +1,69 @@
-        setUpTokens();
-        setUpOracles();
+        realOracleOwner = new ChainlinkOracleOwner();
+        fakeOracleOwner = new ChainlinkOracleOwner();
+        wethAggregator  = new ChainlinkAggregatorLike();
+        oracle          = new ChainlinkOracle(address(wethAggregator), address(1), address(realOracleOwner));
     }
 
     function test_getLatestPrice() public {
         // Assert initial state
-        assertTrue( wethOracle.getLatestPrice() > int256(1));
-        assertTrue(!wethOracle.manualOverride());
-        assertEq(   wethOracle.manualPrice(), int256(0));
+        assertTrue( oracle.getLatestPrice() > int256(1));
+        assertTrue(!oracle.manualOverride());
+
+        assertEq(oracle.manualPrice(), int256(0));
 
         // Try to set manual price before setting the manual override.
-        assertTrue(!securityAdmin.try_setManualPrice(address(wethOracle), int256(45000)));
+        assertTrue(!realOracleOwner.try_chainlinkOracle_setManualPrice(address(oracle), int256(45000)));
 
         // Enable oracle manual override
-        assertTrue(         !cam.try_setManualOverride(address(wethOracle), true));
-        assertTrue(securityAdmin.try_setManualOverride(address(wethOracle), true));
-        assertTrue(   wethOracle.manualOverride());
+        assertTrue(!fakeOracleOwner.try_chainlinkOracle_setManualOverride(address(oracle), true));
+        assertTrue( realOracleOwner.try_chainlinkOracle_setManualOverride(address(oracle), true));
+        assertTrue(oracle.manualOverride());
 
         // Set price manually
-        assertTrue(         !cam.try_setManualPrice(address(wethOracle), int256(45000)));
-        assertTrue(securityAdmin.try_setManualPrice(address(wethOracle), int256(45000)));
-        assertEq(     wethOracle.manualPrice(),    int256(45000));
-        assertEq(     wethOracle.getLatestPrice(), int256(45000));
+        assertTrue(!fakeOracleOwner.try_chainlinkOracle_setManualPrice(address(oracle), int256(45000)));
+        assertTrue( realOracleOwner.try_chainlinkOracle_setManualPrice(address(oracle), int256(45000)));
+        
+        assertEq(oracle.manualPrice(),    int256(45000));
+        assertEq(oracle.getLatestPrice(), int256(45000));
 
         // Change aggregator
-        assertTrue(         !cam.try_changeAggregator(address(wethOracle), 0xb022E2970b3501d8d83eD07912330d178543C1eB));
-        assertTrue(securityAdmin.try_changeAggregator(address(wethOracle), 0xb022E2970b3501d8d83eD07912330d178543C1eB));
-        assertEq(address(wethOracle.priceFeed()),                          0xb022E2970b3501d8d83eD07912330d178543C1eB);
+        assertTrue(!fakeOracleOwner.try_chainlinkOracle_changeAggregator(address(oracle), 0xb022E2970b3501d8d83eD07912330d178543C1eB));
+        assertTrue( realOracleOwner.try_chainlinkOracle_changeAggregator(address(oracle), 0xb022E2970b3501d8d83eD07912330d178543C1eB));
+        
+        assertEq(address(oracle.priceFeed()), 0xb022E2970b3501d8d83eD07912330d178543C1eB);
```

Comment:
Going to let this slide since it was in the original test, but this really should be split out into individual tests for each of these functions.